src/rpc/bootstrap_daemon.h

```diff
@@ -1,26 +1,34 @@
 #pragma  once
 
 #include <functional>
-#include <vector>
+#include <map>
 
 #include <boost/optional/optional.hpp>
+#include <boost/thread/mutex.hpp>
 #include <boost/utility/string_ref.hpp>
 
 #include "net/http_client.h"
 #include "storages/http_abstract_invoke.h"
 
+#include "bootstrap_node_selector.h"
+
 namespace cryptonote
 {
 
   class bootstrap_daemon
   {
   public:
-    bootstrap_daemon(std::function<boost::optional<std::string>()> get_next_public_node);
-    bootstrap_daemon(const std::string &address, const boost::optional<epee::net_utils::http::login> &credentials);
+    bootstrap_daemon(
+      std::function<std::map<std::string, bool>()> get_public_nodes,
```

Comment:
could pass it by const ref, unless there's a good reason not to. I think these objects can get large if they've got capture data (not sure about the details though).

Reply:
Updated

Reply:
Decided to use move semantic instead.